tools/swift-format/Sources/Rules/CommentWhitespace.swift

```diff
@@ -11,5 +11,95 @@ import SwiftSyntax
 ///
 /// - SeeAlso: https://google.github.io/swift#horizontal-whitespace
 public final class CommentWhitespace: SyntaxFormatRule {
+  public override func visit (_ token: TokenSyntax) -> Syntax {
+    var pieces = [TriviaPiece]()
+    var validToken = token
+    var isInvalid = false
+    
+    guard let nextToken = token.nextToken else {
+      // In the case there is a line comment at the end of the file, it ensures
+      // that the line comment has a single space after the `//`.
+      pieces = checksSpacesAfterLineComment(isInvalid: &isInvalid, token: token).reversed()
+      return isInvalid ? token.withLeadingTrivia(Trivia.init(pieces: pieces)) : token
+    }
+    
+    // Ensures the line comment has at least 2 spaces before the `//`.
+    if hasLineComment(trivia: nextToken.leadingTrivia) {
+      let numSpaces = token.trailingTrivia.numberOfSpaces
+      if numSpaces < 2 {
+        isInvalid = true
+        let addSpaces = 2 - numSpaces
+        diagnose(.addSpacesBeforeLineComment(count: addSpaces), on:token)
+        validToken = token.withTrailingTrivia(token.trailingTrivia.appending(.spaces(addSpaces)))
+      }
+    }
+    
+    pieces = checksSpacesAfterLineComment(isInvalid: &isInvalid, token: token).reversed()
+    return isInvalid ? validToken.withLeadingTrivia(Trivia.init(pieces: pieces)) : token
+  }
+  
+  /// Returns a boolean indicating if the given trivia contains a line comment.
+  private func hasLineComment (trivia: Trivia) -> Bool {
+    // When the comment isn't inline with code, it doesn't need to
+    // to check that there are two spaces before the line comment.
+    if let firstPiece = trivia.reversed().last {
+      if case .newlines(_) = firstPiece {
+        return false
+      }
+    }
+    for piece in trivia.reversed() {
+      if case .lineComment(_) = piece {
+        return true
+      }
+    }
+    return false
+  }
+  
+  /// Ensures the line comment has exactly one space after `//`.
+  private func checksSpacesAfterLineComment(isInvalid: inout Bool, token: TokenSyntax) -> [TriviaPiece] {
+    var pieces = [TriviaPiece]()
+    
+    for piece in token.leadingTrivia.reversed() {
+      // Checks if the line comment piece follows the right format,
+      // if it doesn't it modifies the comment to the right form.
+      if case .lineComment(var text) = piece,
+         invalidLineComment(textLineComment: &text, token: token) {
+        isInvalid = true
+        pieces.append(TriviaPiece.lineComment(text))
+      }
+      else {
+        pieces.append(piece)
+      }
+    }
+    return pieces
+  }
+  
+  /// Returns a boolean indicating if the comment had an invalid format,
+  /// if it does, the text of the comment is modified to the correct format.
+  private func invalidLineComment (textLineComment: inout String, token: TokenSyntax) -> Bool {
```

Comment:
`invalidLineComment` doesn't give any hints about what the function does; maybe `formatLineComment` instead?